### PR TITLE
Remove url port for Geolocation

### PR DESCRIPTION
**Pull request by superpoussin22**
quick-fix

### Describe the Pull Request

```
if url contain a port the geolocation did not work 
quick fix 

```

### Checklist for PR

- [X] Run MobSF unit tests and lint `tox -e lint,test`.
- [X] Tested Working on Linux, Mac, Windows, and Docker
- [X] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)

### Additional Comments (if any)

```
DESCRIBE HERE
```

### CODE REVIEW

File: MalwareAnalyzer/views/domain_check.py [hunk 1]
```diff
@@ -147,6 +147,8 @@ def get_domains(urls):
         domains = []
         for url in urls:
             parsed_uri = urlparse(url)
+            if urlparse(url).port:
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/domain_check.py]: Change to `parsed_uri.port`

File: MalwareAnalyzer/views/domain_check.py [hunk 2]
```diff
@@ -147,6 +147,8 @@ def get_domains(urls):
...
+                url = urlparse(url).hostname
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/domain_check.py]: Change to `parsed_uri.hostname`

File: MalwareAnalyzer/views/domain_check.py [hunk 3]
```diff
@@ -147,6 +147,8 @@ def get_domains(urls):
...
             if not urlparse(url).scheme:
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/domain_check.py]: `parsed_uri.scheme `

Comment (superpoussin22) [reply MalwareAnalyzer/views/domain_check.py]: if doing this the url will not be parsed after removing the port :( need to rethink it

File: MalwareAnalyzer/views/domain_check.py [hunk 4]
```diff
@@ -130,11 +130,11 @@ def get_netloc(url):
...
         domain = ''
-        parsed_uri = urlparse(url)
-        if not urlparse(url).scheme:
+        parse_uri = urlparse(url)
+        if not parse_uri.scheme:
             url = '//' + url
-            parsed_uri = urlparse(url)
-            domain = '{uri.netloc}'.format(uri=parsed_uri)
+            parse_uri = urlparse(url)
+            domain = '{uri.netloc}'.format(uri=parse_uri)
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/domain_check.py]: Is this intent correct?

File: MalwareAnalyzer/views/domain_check.py [hunk 5]
```diff
@@ -146,13 +146,11 @@ def get_domains(urls):
...
         for url in urls:
-            parsed_uri = urlparse(url)
-            if parsed_uri.port:
-                url = parsed_uri.hostname
-            if parsed_uri.port or not parsed_uri.scheme:
-                url = '//' + url
-                parsed_uri = urlparse(url)
-            domain = '{uri.netloc}'.format(uri=parsed_uri)
+            parse_uri = urlparse(url)
+            if (parse_uri.port or not parse_uri.scheme) and parse_uri.hostname:
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/domain_check.py]: why explicit check for `parse_uri.hostname`?  At this level it already after URL extraction, it must have a hostname?

File: MalwareAnalyzer/views/domain_check.py (hunk 4 above)
Comment (ajinabraham) [reply MalwareAnalyzer/views/domain_check.py]: The domain should be available if `scheme` is present as well right?

File: MalwareAnalyzer/views/domain_check.py (hunk 5 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/domain_check.py]: I see sometime doamin with empty value which will make the function crash, don't know why it is possible, so I prefer add this test to secure the function

File: MalwareAnalyzer/views/domain_check.py (hunk 4 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/domain_check.py]: I replace parsed with parse to be consistent with the check_domain function 
you can have domain without scheme and with scheme this is not a problem

File: MalwareAnalyzer/views/domain_check.py (hunk 5 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/domain_check.py]: in google one apk you have one malformed  url which reports empty scheme, empty netloc and path=www.google.com/visualsearch/lh/" and that's make the function crash if hostname is not checked

File: MalwareAnalyzer/views/domain_check.py (hunk 4 above)
Comment (ajinabraham) [reply MalwareAnalyzer/views/domain_check.py]: The variable `domain` is assigned only when scheme is not present.

See: https://github.com/MobSF/Mobile-Security-Framework-MobSF/blob/bbce1323ba0b5e6f8bd29ad1b581a47cffa0cda1/MalwareAnalyzer/views/domain_check.py#L137

 I think domain should have indentation of 4 spaces removed.

```
if not parse_uri.scheme:
            url = '//' + url
            parse_uri = urlparse(url)
 domain = '{uri.netloc}'.format(uri=parse_uri)
```

File: MalwareAnalyzer/views/domain_check.py (hunk 5 above)
Comment (ajinabraham) [reply MalwareAnalyzer/views/domain_check.py]: okey. sounds good.

**Comment by ajinabraham** [comment]
@superpoussin22 I have added my comments

File: MalwareAnalyzer/views/domain_check.py (hunk 4 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/domain_check.py]: simplify the code

**Pull request merged**